### PR TITLE
deps: update for https://github.com/celestiaorg/celestia-node/pull/1913 and v0.8.2 and common docker pipeline

### LINKED ISSUE #17: Edit to reflect changes in docker on celestia-node #1913
Error building this when testing `celestia-node v0.8.0-rc1`, needs to be updated to reflect https://github.com/celestiaorg/celestia-node/pull/1913

```sh
docker build . -t celestia-local-devnet

[+] Building 0.5s (8/11)
 => [internal] load build definition from Dockerfile                    0.0s
 => => transferring dockerfile: 32B                                     0.0s
 => [internal] load .dockerignore                                       0.0s
 => => transferring context: 2B                                         0.0s
 => [internal] load metadata for ghcr.io/celestiaorg/celestia-app:v0.1  0.4s
 => [internal] load metadata for ghcr.io/celestiaorg/celestia-node:v0.  0.4s
 => CACHED [celestia-node 1/1] FROM ghcr.io/celestiaorg/celestia-node:  0.0s
 => [stage-1 1/5] FROM ghcr.io/celestiaorg/celestia-app:v0.12.1@sha256  0.0s
 => CANCELED [internal] load build context                              0.0s
 => => transferring context:                                            0.0s
 => ERROR [stage-1 2/5] COPY --from=celestia-node /celestia /           0.0s
------
 > [stage-1 2/5] COPY --from=celestia-node /celestia /:
------
ERROR: failed to solve: failed to compute cache key: "/celestia" not found: not found
```

### PR DESCRIPTION
**Pull request by jcstein**
## Overview

https://github.com/celestiaorg/celestia-node/pull/1913
Resolves #17 and #11 

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Comment by jcstein** [comment]
Logs from run with v0.8.0-rc2: https://app.warp.dev/block/eafrK0V7JZaJseW42OnOmz

**Review by nashqueue — APPROVED** [review]

**Review by tuxcanfly — APPROVED** [review]
OK.

**Comment by jcstein** [comment]
@jrmanes - accidentally committed to my fork. adds a common CI pipeline here too

**Review by jrmanes — APPROVED** [review]
LGTM! 🚀 
thanks for making the time! 😊

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,8 @@
-FROM ghcr.io/celestiaorg/celestia-node:v0.7.2 AS celestia-node
+FROM ghcr.io/celestiaorg/celestia-node:v0.8.1 AS celestia-node
 
 FROM ghcr.io/celestiaorg/celestia-app:v0.12.1
```
Comment (Bidon15) [inline Dockerfile:3]: should be v0.12.2 according to go.mod from v0.8.1 node 
cc: @celestiaorg/devops we need to think about how to maintain this zoo of dockerfiles tbh

Comment (jcstein) [reply Dockerfile:3]: it still fails

```bash
jcs @ ~/DEV/local-celestia-devnet % docker build . -t celestia-local-devnet-8-1
[+] Building 0.7s (4/4) FINISHED                                                                                                                                       
 => [internal] load build definition from Dockerfile                                                                                                              0.0s
 => => transferring dockerfile: 359B                                                                                                                              0.0s
 => [internal] load .dockerignore                                                                                                                                 0.0s
 => => transferring context: 2B                                                                                                                                   0.0s
 => CANCELED [internal] load metadata for ghcr.io/celestiaorg/celestia-app:v0.12.2                                                                                0.6s
 => ERROR [internal] load metadata for ghcr.io/celestiaorg/celestia-node:v0.8.1                                                                                   0.6s
------
 > [internal] load metadata for ghcr.io/celestiaorg/celestia-node:v0.8.1:
------
ERROR: failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to create LLB definition: no match for platform in manifest sha256:1705bb2d5e8a0efa797db6964003c47c7a8cb452a1c92c3f4a9302390cb9f858: not found
```

**Comment by jcstein** [comment]
Yeah, I know, I only tagged node because error was related. Thanks for picking this up 🙏

File: .github/workflows/docker-build-publish.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      - "v[0-9]+.[0-9]+.[0-9]+-beta.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+"
+  pull_request:
+
+jobs:
+  docker-security-build:
+    permissions:
+      contents: write
+      packages: write
+    uses: celestiaorg/.github/.github/workflows/reusable_dockerfile_pipeline.yml@main # yamllint disable-line rule:line-length
```
Comment (smuu) [inline .github/workflows/docker-build-publish.yml:20]: This will only push on version tags, which we don't have in this repo.

Comment (jcstein) [reply .github/workflows/docker-build-publish.yml:20]: @jrmanes - could you please help me out here? also, i'm confused as to why we'd move this to rollkit and then use celestia workflows 🤔

File: Dockerfile (hunk 1 above)
Comment (jcstein) [reply Dockerfile]: yay. it ran on 0.8.2

File: .github/workflows/docker-build-publish.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jcstein) [reply .github/workflows/docker-build-publish.yml:20]: ok. done

Comment (jrmanes) [reply .github/workflows/docker-build-publish.yml]: hello @jcstein
atm we only have these common pipelines in Celestiaorg, though @smuu is working on it for creating them also in Rollkit. Once we'll have them there, we will have to change the reference to rollkit repo 👌 
hope i've solved you the doubts :)

Comment (jcstein) [reply .github/workflows/docker-build-publish.yml]: okay, thank you

Comment (jcstein) [reply .github/workflows/docker-build-publish.yml]: okay. so what do i need to do here? there is a "main" branch on this repo?

**Review by Bidon15 — APPROVED** [review]

**Review by nashqueue — APPROVED** [review]
Proxy Approve from viet

File: .github/workflows/docker-build-publish.yml (hunk 2 above)
Comment (Bidon15) [reply .github/workflows/docker-build-publish.yml]: we will need versioning for this @smuu in the future. 
As all the docs of rollkit mentions how to run tutorials on different celestia testnets. 

Hence why, we also need to replicate some versioning here to not get lost in the weeds.

Comment (smuu) [reply .github/workflows/docker-build-publish.yml]: To verify: Does this ticket track your idea @Bidon15?
https://github.com/celestiaorg/devops/issues/174

**Pull request merged**